File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-0.js
class Solution {
    /**
     * @param {string} s
     * @return {number}
     */
    lengthOfLongestSubstring(s) {
       let map = new Map()
       let res = 0
       let l =0
       for(let r=0; r < s.length; r++){
            let char = s[r]
            if(map.has(char)){
                l = Math.max(map.get(char)+1, l)
            }
         map.set(char, r)
         res = Math.max(res, (r-l)+1)
       }
       return res
    }
}

File: Data Structures & Algorithms/eating-bananas/submission-0.java
class Solution {
    public int minEatingSpeed(int[] piles, int h) {
        int left = 1;
        int right = Arrays.stream(piles).max().getAsInt();
        int res = right;

        while(left <= right){
            int m = (left + right)/2;
            int value = 0;
            for(int num : piles){
                value += Math.ceil((double) num / m);
            }

            if(value <= h){
                res = m;
                right = m-1;
            }
            else {
                left = m+1;
            }
        }

        return res;
    }
}
